scipy/linalg/_special_matrices.py

```diff
@@ -109,13 +109,29 @@ def circulant(c):
            [2, 1, 3],
            [3, 2, 1]])
 
+    >>> circulant([[1, 2, 3], [4, 5, 6]])
+    array([[[1, 3, 2],
+            [2, 1, 3],
+            [3, 2, 1]],
+           [[4, 6, 5],
+            [5, 4, 6],
+            [6, 5, 4]]])
     """
-    c = np.asarray(c).ravel()
+    c = np.atleast_1d(c)
```

Comment:
`ravel` would turn 0d arrays into 1d arrays, so I preserve that.

Reply:
Looks like the discussion below concluded that this was ok